cms/server/contest/handlers/contest.py

```diff
@@ -324,7 +330,8 @@ def setup_locale(self):
             lang_code = self.browser_lang
 
         self.set_header("Content-Language", lang_code)
-        self.translation = self.all_translations[lang_code.replace("-", "_")]
+        self.translation = \
+            self.available_translations[lang_code.replace("-", "_")]
```

Comment:
Nit/ or in another commit - if cookie lang is in lang_codes, we could save looking at headers and parsing and matching.

Reply:
Nevermind, I see that automated is used in the rendering regardless.